File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.py
class Solution(object):
    def check(self, nums):
        dummy = nums[:]
        dummy.sort()
        length = len(nums)
        for i in range(length):
            flag = True
            first = dummy[0]
            for j in range(length - 1):
                dummy[j] = dummy[j+1]
            dummy[-1] = first
            for k in range(length):
                if(nums[k]!=dummy[k]):
                    flag = False
                    break
            if(flag is True):
                return True
        return False
        
        

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        if(n==1) return true;
        int cnt = 1;
       
        for(int i = 1 ; i<n*2;i++){
            if(nums[i%n]>=nums[(i-1)%n]){
                cnt+=1;
            }else{
                cnt=1;
            }
            if(cnt==nums.length) return true;
            
        }
       
        return false;

    }
}